tests/foreman/api/test_rhci.py

```diff
@@ -0,0 +1,125 @@
+
+        @Assert: An RHCI deployment is created with a random name.
+
+        @Feature: RHCI Deployment
+
+        """
+        deployment = entities.RHCIDeployment(
+            name=data,
+            organization=self.org['organization']['id'],
+            lifecycle_environment=self.lc_env.id,
+            deploy_rhev=True,
+            rhev_engine_host=self.rhev_engine_host.id,
+            rhev_storage_type='NFS',
+            rhev_engine_admin_password='changeme',
+        ).create()
+        self.assertEquals(deployment.name, data)
+        self.assertEqual(
+            deployment.organization.id, self.org['organization']['id'])
+        self.assertEqual(deployment.lifecycle_environment.id, self.lc_env.id)
+        self.to_delete.append(deployment)
+
+    @data(
+        {'name': gen_string(str_type='alpha'),
+         'new-name': gen_string(str_type='alpha')},
+        {'name': gen_string(str_type='alphanumeric'),
+         'new-name': gen_string(str_type='alphanumeric')},
+        {'name': gen_string(str_type='cjk'),
+         'new-name': gen_string(str_type='cjk')},
+        {'name': gen_string(str_type='latin1'),
+         'new-name': gen_string(str_type='latin1')},
+    )
+    def test_positive_update_name(self, data):
+        """@Test: Update a deployment's name.
+
+        @Assert: An RHCI deployment is updated with a random name.
+
+        @Feature: RHCI Deployment
+
+        """
+        deployment = entities.RHCIDeployment(
+            name=data['name'],
+            organization=self.org['organization']['id'],
+            lifecycle_environment=self.lc_env.id,
+            deploy_rhev=True,
+            rhev_engine_host=self.rhev_engine_host.id,
+            rhev_storage_type='NFS',
+            rhev_engine_admin_password='changeme',
+        ).create()
+        self.assertEquals(deployment.name, data['name'])
+
+        # perform the update
+        deployment.name = data['new-name']
+        deployment = deployment.update(['name'])
+        self.assertEqual(deployment.name, data['new-name'])
+        self.to_delete.append(deployment)
+
+    @data(
+        gen_string(str_type='alpha'),
+    )
+    def test_positive_delete_deployment(self, data):
+        """@Test: Create and delete a simple deployment.
+
+        @Assert: An RHCI deployment is deleted.
+
+        @Feature: RHCI Deployment
+
+        """
+        deployment = entities.RHCIDeployment(
+            name=data,
+            organization=self.org['organization']['id'],
+            lifecycle_environment=self.lc_env.id,
+            deploy_rhev=True,
+            rhev_engine_host=self.rhev_engine_host.id,
+            rhev_storage_type='NFS',
+            rhev_engine_admin_password='changeme',
+        ).create()
+        self.assertEquals(deployment.name, data)
+
+        # Perform the delete
+        deployment.delete()
+
+        # Check to make sure the deployment has been deleted
+        with self.assertRaises(HTTPError):
+            deployment.read()
```

Comment:
This works fine. For the sake of teaching, I want to point out that you can also write this:

```python
self.assertEqual(deployment.read_raw().status_code, httplib.NOT_FOUND)  # 404
```

But this assumes that the server returns a 404 instead of a 400 or something else.